File: backend/src/main/java/com/routinesystem/backend/services/ExerciseService.java
package com.routinesystem.backend.services;

import com.routinesystem.backend.dtos.ExerciseDto;

import java.util.List;

public interface ExerciseService {
    ExerciseDto createExercise(ExerciseDto exerciseDto);
    ExerciseDto getExerciseById(Long exerciseId);
    List<ExerciseDto> getAllExercises();
    ExerciseDto updateExercise(Long exerciseId, ExerciseDto exerciseDto);
    void deleteExercise(Long exerciseId);
}

File: frontend/src/services/ExerciseService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/api/exercises";

class ExerciseService {

    getAllExercises() {
        return axios.get(BASE_URL);
    }

    createExercise(exercise){
        return axios.post(BASE_URL, exercise);
    }

    getExerciseById(exerciseId){
        return axios.get(`${BASE_URL}/${exerciseId}`);
    }

    updateExercise(exerciseId, exercise){
        return axios.put(`${BASE_URL}/${exerciseId}`, exercise);
    }

    deleteExercise(exerciseId){
        return axios.delete(`${BASE_URL}/${exerciseId}`);
    }

    getMuscles(){
        return axios.get(`${BASE_URL}/muscles`);
    }

}

export default new ExerciseService();
